kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/client/DefaultJsonHandlerSuite.scala

```diff
@@ -346,6 +350,157 @@ class DefaultJsonHandlerSuite extends AnyFunSuite with TestUtils with VectorTest
     assert(e.getMessage.contains("Could not parse JSON"))
   }
 
+  test("read json files") {
+    val testFiles = fsClient.listFrom(getTestResourceFilePath("json-files/1.json"))
+
+    val actPaths = new ArrayBuffer[String]()
+    val actSizes = new ArrayBuffer[Long]()
+    val actDataChanges = new ArrayBuffer[Boolean]()
+    val data = jsonHandler.readJsonFiles(
+      testFiles,
+      new StructType()
+        .add("path", StringType.STRING)
+        .add("size", LongType.LONG)
+        .add("dataChange", BooleanType.BOOLEAN),
+      Optional.empty()
+    )
+    try {
+      data.forEach(batch => {
+        val dataBatchRows = batch.getRows
+        try {
+          dataBatchRows.forEach(row => {
+            actPaths += row.getString(0)
+            actSizes += row.getLong(1)
+            actDataChanges += row.getBoolean(2)
+          })
+        } finally {
+          dataBatchRows.close()
+        }
+      })
+    } finally {
+      data.close()
+    }
+
```

Comment:
@allisonport-db has developed a neat Kernel test framework in Scala where the comparisons between the expected and actual rows are done using just a few lines.

Convert the act tests results to `TestRow`:
```
val actResult =jsonHandler.readJsonFiles.toSeq.map(TestRow(_))
```

Create `TestRow` objects for expected result using
```
val expResult = Seq(
   TestRow("part-00000-d83dafd8-c344-49f0-ab1c-acd944e32493-c000.snappy.parquet", 348L, true),
   TestRow("part-00000-cb078bc1-0aeb-46ed-9cf8-74a843b32c8c-c000.snappy.parquet", 687L, true)
   ... for the rest of expected rows ...
```

Now compare
```
assert(actResult === expResult)
```